spec/OpenLineage.yml

```diff
@@ -0,0 +1,212 @@
+openapi: 3.0.2
+info:
+  title: OpenLineage
+  version: 0.0.1
+  description: OpenLineage is an open source **lineage and metadata collection API** for the data ecosystem.
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+paths:
+  /runs:
+    post:
+      summary: Updates the state of a run
+      description: Updates a run state for a job.
+      operationId: postRunStateUpdate
+      tags:
+        - OpenLineage
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RunStateUpdate'
+      responses:
+        '200':
+          description: OK
+
+components:
+  schemas:
+    RunStateUpdate:
+      type: object
+      properties:
+        id:
+          description: the unique identifier for that run
+          type: string
+          format: uuid
+        transition:
```

Comment:
It may be nice to record multiple state transitions in a single event since the relationship of `runs` -> `run state` is 1-n. However it changes the relationship of an event with the dataset.

Reply:
I was thinking we could have an endpoint where we send an array of RunStateUpdate. The idea being that the job and dataset are described as observed at the time of the transition. If nothing has changed or we didn't observe it we can have a RunStateUpdate event that contains only the transition information.